### PR TITLE
Explorer UI - adjusted UI to use new HTTP routes for /connections

### PR DESCRIPTION
**Pull request by thjaeckle**
* also added an example template using a JS payload mapper

To be merged after #1406 is done.

Fixes: #1441 

@thfries would be cool if you could have a look on the changes

### CODE REVIEW

File: ui/modules/connections/connections.js [hunk 1]
```diff
@@ -257,16 +257,16 @@ function loadConnections() {
...
-        row.insertCell(0).innerHTML = connection.name;
-      } else {
+      if (API.env() === 'ditto_2') {
         API.callConnectionsAPI('retrieveConnection', (dittoConnection) => {
-          row.insertCell(0).innerHTML = dittoConnection.name;
-        },
-        id);
+           row.insertCell(0).innerHTML = dittoConnection.name;
+         },
+         id);
```
Comment (thfries) [inline ui/modules/connections/connections.js:267]: Indentation was proclaimed by eslint
```suggestion
         row.insertCell(0).innerHTML = dittoConnection.name;
       },
       id);
```

**Review by thfries — CHANGES_REQUESTED** [review]
Thanks for doing this, @thjaeckle.
LGTM. The new ditto 3 connection API is not yet on master, so I can not test it?
I created an input field for the new ditto version and tried to provide it as a patch here. Does that work?
[versionFieldPatch.txt](https://github.com/eclipse/ditto/files/9591676/versionFieldPatch.txt)

**Comment by thjaeckle** [comment]
> Thanks for doing this, @thjaeckle. LGTM. The new ditto 3 connection API is not yet on master, so I can not test it? I created an input field for the new ditto version and tried to provide it as a patch here. Does that work? [versionFieldPatch.txt](https://github.com/eclipse/ditto/files/9591676/versionFieldPatch.txt)

The HTTP API is not yet merged - we think it however will be this week.
I just wanted to provide the adjustments so that we quickly can move forward releasing Ditto 3.0.

I included your patch with the input field for the Ditto version, thx 👍

File: ui/modules/environments/environments.html [hunk 2]
```diff
@@ -51,6 +51,10 @@ <h5>Environments</h5>
                         title="URI including protocol, host and port (e.g. http://localhost:8080)">API URI</label>
                     <input type="text" class="form-control form-control-sm" id="inputApiUri"></input>
                 </div>
+                <div class="input-group input-group-sm mt-1">
+                    <label class="input-group-text">Ditto Version</label>
+                    <input type="integer" class="form-control form-control-sm" id="inputDittoVersion"></input>
```
Comment (thjaeckle) [inline ui/modules/environments/environments.html:56]:
```suggestion
                    <input type="number" class="form-control form-control-sm" id="inputDittoVersion"></input>
```

**Comment by thfries** [comment]
Nice! I like all the changes. Thanks for all the improvements!

File: ui/modules/things/things.html [hunk 3]
```diff
@@ -88,7 +88,7 @@ <h5>Things</h5>
                                 <div class="invalid-feedback"></div>
                             </div>
                             <div class="input-group input-group-sm mb-1">
-                                <label class="input-group-text">Definition</label>
+                                <label class="input-group-text">Definition template</label>
```
Comment (thfries) [inline ui/modules/things/things.html:91]: I would stick to just "Definition", as it represents the definition of the thing (if you select a thing, the definition is filled into that field).

Comment (thjaeckle) [reply ui/modules/things/things.html:91]: Ok, agreed, I will undo this. The "template" mechanism with the dropdown at this place however does not have much to do with the "definition" in my opinion.

Comment (thjaeckle) [inline ui/modules/things/things.html:91]:
```suggestion
                                <label class="input-group-text">Definition</label>
```

**Pull request merged**